catalog/spatial/geocoding/spatial-geocoding-offline-catalog/src/test/java/org/codice/ddf/spatial/geocoding/index/GeoNamesCatalogIndexerTest.java

```diff
@@ -272,18 +306,112 @@ public void testUpdateIndexSourceUnavailableException() throws Exception {
   @Test
   public void testUpdateIndexWithGeoEntryList() throws Exception {
     List<GeoEntry> geoEntries = Collections.singletonList(GEO_ENTRY);
-    geoNamesCatalogIndexer.updateIndex(geoEntries, false, progressCallback);
+    geoNamesCatalogIndexer.updateIndex(geoEntries, false, progressCallback, GEO_ENTRY_DATA_SOURCE);
     verify(catalogFramework, times(1)).create(any(CreateRequest.class));
     verify(catalogProvider, times(0)).delete(any(DeleteRequest.class));
     verify(catalogFramework, times(0)).query(any(QueryRequest.class));
   }
 
   @Test
   public void testUpdateIndexWithGeoEntryListWithCreate() throws Exception {
+    setupTestForCreate();
+
     List<GeoEntry> geoEntries = Collections.singletonList(GEO_ENTRY);
-    geoNamesCatalogIndexer.updateIndex(geoEntries, true, progressCallback);
+    geoNamesCatalogIndexer.updateIndex(geoEntries, true, progressCallback, GEO_ENTRY_DATA_SOURCE);
     verify(catalogFramework, times(1)).create(any(CreateRequest.class));
     verify(catalogProvider, times(1)).delete(any(DeleteRequest.class));
-    verify(catalogFramework, times(1)).query(any(QueryRequest.class));
+    verify(catalogFramework, times(2)).query(any(QueryRequest.class));
+  }
+
+  private void setupTestForCreate() throws Exception {
+    queryResponse = mock(QueryResponse.class);
+    when(queryResponse.getResults()).thenReturn(Collections.emptyList());
+    when(catalogFramework.query(any(QueryRequest.class)))
+        .thenReturn(
+            new TestQueryResponse(Collections.singletonList(new ResultImpl(new MetacardImpl()))));
+  }
+
+  private class TestQueryResponse implements QueryResponse {
```

Comment:
✏️ is it possible to mock this instead?  it might clean up this test class

Reply:
The intent here was to let it work one way the first call, and after that to work a different way in order to simulate the data being removed. I couldn't think of a way to do that with a mock.

Reply:
You can achieve the same effect with mocks. Something like this: 
```
when(mockResponse.getResults()).thenReturn(expectedResults).thenReturn(Collections.emptyList());
```